src/logic/CompetitionOperator.java

```diff
@@ -0,0 +1,98 @@
+package logic;
+
+import data.Competition;
+import org.apache.commons.lang3.math.NumberUtils;
+
+import java.util.Scanner;
+
+public class CompetitionOperator {
+    private final static String FINISH_COMPETITIONS = "STOP";
+    private final static String SORT_DESCENDING = "1";
+    private final static String SORT_ASCENDING = "2";
+
+    public static void run(Competition competition) {
+        boolean reverseList = false;
+        boolean loop = true;
+        Scanner scanner = new Scanner(System.in);
+
+        do{
+            System.out.println("Wybierz tryb sortowania wyników\n" +
+                    "1. Malejąco\n" +
+                    "2. Rosnąco");
+            String sortOption = scanner.nextLine();
+            switch(sortOption.toUpperCase()) {
+                case SORT_DESCENDING -> {
+                    System.out.println("Wybrano malejące sortowanie wyników");
+                    loop = false;
+                }
+                case SORT_ASCENDING -> {
+                    System.out.println("Wybrano rosnące sortowanie wyników");
+                    reverseList = true;
+                    loop = false;
+                }
+                default -> {
+                    System.out.println("Wybrano zły numer: " + sortOption);
+                }
+            }
+        } while (loop);
+
+        loop = true; //dla następnej pętli
+        do {
+            System.out.println("Podaj wynik kolejnego gracza (lub stop):");
+
+            String playerInfo = scanner.nextLine();
+            if(playerInfo.toUpperCase().equals(FINISH_COMPETITIONS)) {
+                loop = false;
+                // ---- test block ----
+                competition.addPlayerResult("Kamil", "Zalwert", 123);
+                competition.addPlayerResult("Maciej", "Agrafka", 77);
+                competition.addPlayerResult("Jan", "Kowalski", 56);
+                competition.addPlayerResult("Paweł", "Pierwszy", 34);
+                competition.addPlayerResult("Andrzej", "Drugi", 12);
+                competition.addPlayerResult("Andrzej", "Agrafka", 77);
+                // ---- end of test ----
+                ReportCreator.generateReport("stats.csv", competition.printListOfPlayersInCsvFormat(reverseList));
```

Comment:
generacja raportu jest wewnatrz petli ktora zczytuje dane od uzytkownika. Mimo ze to zadziala bo po dkoniec bloku jest `return` to jest to nieczytelne